nicegui/elements/anywidget/anywidget.js

```diff
@@ -0,0 +1,129 @@
+  template: "<div></div>",
+  mounted() {
+    this.init_widget();
+  },
+  methods: {
+    _log(...args) {
+      if (this._debug) {
+        console.log("NiceGUI-Anywidget", ...args);
+      }
+    },
+    init_widget() {
+      (async () => {
+        const emit_to_py = this.$emit;
+        const log = this._log;
+
+        // Implement AFM: https://anywidget.dev/en/afm/
+        // References:
+        // * Marimo AFM impl:
+        // https://github.com/marimo-team/marimo/blob/7f3023ff0caef22b2bf4c1b5a18ad1899bd40fa3/frontend/src/plugins/impl/anywidget/AnyWidgetPlugin.tsx#L161-L267
+        const model = {
+          attributes: { ...this.traits },
+          callbacks: {},
+          get: function (key) {
+            log('Getting value for', key, ':', this.attributes[key]);
+            const value = this.attributes[key];
+            try {
+              // TODO: this should not be necessary but was running into some
+              // JavaScript issues that haven't tried to figure out
+              return JSON.parse(JSON.stringify(value));
+            } catch (e) {
+              // If value is not serializable, return null or a fallback
+              console.warn('NiceGUI-Anywidget: Value for key', key, 'is not JSON-serializable:', value);
+              return null;
+            }
+          },
+          set: function (key, value) {
+            log('Setting value for', key, ':', value);
+            this.attributes[key] = value;
+            this.emit('change:' + key, value);
+          },
+          save_changes: function () {
+            log('Saving changes:', this.attributes);
+
+            // Trigger any change callbacks
+            if (this.callbacks['change'] && Array.isArray(this.callbacks['change'])) {
+              this.callbacks['change'].forEach((cb) => cb());
+            }
+
+            // Propagate the change back to python backend;
+            // currently serializing all traits instead of just the changed ones
+            // (ideally would do this to reduce communication overhead)
+            emit_to_py('update:traits', { ...this.attributes });
+          },
+          on: function (event, callback) {
+            log('Registering callback for event:', event);
+            if (!this.callbacks[event]) {
+              this.callbacks[event] = [];
+            }
+            this.callbacks[event].push(callback);
+          },
+          off: function (event, callback) {
+            if (!event) {
+              this.callbacks = {};
+              return;
+            }
+            if (!callback) {
+              this.callbacks[event] = [];
+              return;
+            }
+            this.callbacks[event]?.delete(callback);
+          },
+          emit: function (event, value) {
+            if (this.callbacks[event]) {
+              this.callbacks[event].forEach(cb => cb(value));
+            }
+          },
+          send: function (content, callbacks, buffers) {
+            if (buffers) {
```

Comment:
After looking at the `ipywidgets` docs, I found out that the python widget receives this in a callback registered by `on_msg()` so it seems like it should not be too hard to handle JS -> Python arbitrary communication.  https://ipywidgets.readthedocs.io/en/latest/_modules/ipywidgets/widgets/widget.html#Widget.on_msg 

Conversely for Python -> JS though I am not sure what one would do on an arbitrary `AnyWidget.send()`: https://ipywidgets.readthedocs.io/en/latest/_modules/ipywidgets/widgets/widget.html#Widget.send I don't see an obvious way to intercept the `send()` callback and redirect it through NiceGUI comms instead.